pkg/util/encoding/encoding.go

```diff
@@ -972,6 +976,63 @@ func decodeTime(b []byte) (r []byte, sec int64, nsec int64, err error) {
 	return b, sec, nsec, nil
 }
 
+// EncodeGeoAscending encodes a geopb.SpatialObject value in ascending order and
+// returns the new buffer.
+// TODO(otan): this should ideally just be encoded by {SRID,Shape,Raw Points}.
+// EWKB is expensive to encode. However, we don't store this as a PRIMARY KEY
+// (this is needed for GROUP BY only for now), so we ignore it for now.
+func EncodeGeoAscending(b []byte, g *geopb.SpatialObject) ([]byte, error) {
```

Comment:
As with my comment above, why doesn't this just use encodeBytesAscending?

Reply:
the marshal / unmarshal abstraction is at the very least helpful in this situation.